src/main/java/com/puppycrawl/tools/checkstyle/utils/CodePointUtil.java

```diff
@@ -42,4 +42,24 @@ public static boolean isBlank(int... codePoints) {
                 .allMatch(Character::isWhitespace);
     }
 
+    /**
+     * Checks if the given code point array contains only whitespace up to specified index.
+     *
+     * @param codePoints
+     *            array of Unicode code point of string to check
+     * @param index
+     *            index to check up to (exclusive)
+     * @return true if all code points preceding given index are whitespace
+     */
+    public static boolean hasWhitespaceBefore(int index, int... codePoints) {
+        boolean result = true;
+        for (int i = 0; i < index; i++) {
+            if (!CommonUtil.isCodePointWhitespace(codePoints, i)) {
+                result = false;
+                break;
+            }
+        }
+        return result;
```

Comment:
I suggest to change it to use Arrays.stream, like
```
Arrays.stream(codepoints, 0, index).allMatch(CommonUtil::isCodePointWhitespace)
```

Reply:
changed to `Arrays.stream(codePoints, 0, index).allMatch(codePoint -> Character.isWhitespace(Character.toChars(codePoint)[0]))`
Since, isCodePointWhiteSpace takes array and index, cannot use method reference.